From 38e4f49b0e523544f2e96f64eff28f69848f6aba Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Mon, 15 Sep 2014 11:32:40 +0800
Subject: [PATCH] ENGR00317981 ARM: dts: imx: apply ENET IRQ workaround for
 sabresd board

This a forward porting of commit (ENGR00313685-15 ARM: dts: imx: apply
ENET IRQ workaround for sabresd board) from imx_3.10.y to imx_3.14.y.

Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 arch/arm/boot/dts/Makefile                   |  1 +
 arch/arm/boot/dts/imx6dl-sabresd-enetirq.dts | 18 ++++++++++++++++++
 arch/arm/boot/dts/imx6q-sabresd-enetirq.dts  |  9 +++++++++
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi       |  6 ++++++
 4 files changed, 34 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b8fcad8..8094ced 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -158,6 +158,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6dl-sabreauto.dtb \
 	imx6dl-sabreauto-flexcan1.dtb \
 	imx6dl-sabresd.dtb \
+	imx6dl-sabresd-enetirq.dtb \
 	imx6dl-sabresd-hdcp.dtb \
 	imx6dl-wandboard.dtb \
 	imx6q-arm2.dtb \
diff --git a/arch/arm/boot/dts/imx6dl-sabresd-enetirq.dts b/arch/arm/boot/dts/imx6dl-sabresd-enetirq.dts
new file mode 100644
index 0000000..b23d0eb
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-sabresd-enetirq.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dl-sabresd.dts"
+
+&fec {
+	pinctrl-0 = <&pinctrl_enet &pinctrl_enet_irq>;
+	interrupts-extended = <&gpio1 6 0x04>, <&intc 0 119 0x04>;
+};
+
+&i2c3 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6q-sabresd-enetirq.dts b/arch/arm/boot/dts/imx6q-sabresd-enetirq.dts
index 6dda35c..14b1cfa3 100644
--- a/arch/arm/boot/dts/imx6q-sabresd-enetirq.dts
+++ b/arch/arm/boot/dts/imx6q-sabresd-enetirq.dts
@@ -7,3 +7,12 @@
  */
 
 #include "imx6q-sabresd.dts"
+
+&fec {
+	pinctrl-0 = <&pinctrl_enet &pinctrl_enet_irq>;
+	interrupts-extended = <&gpio1 6 0x04>, <&intc 0 119 0x04>;
+};
+
+&i2c3 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 8a710cc..e761df6 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -380,6 +380,12 @@
 			>;
 		};
 
+		pinctrl_enet_irq: enetirqgrp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_6__ENET_IRQ		0x000b1
+			>;
+		};
+
 		pinctrl_gpio_keys: gpio_keysgrp {
 			fsl,pins = <
 				MX6QDL_PAD_EIM_D29__GPIO3_IO29 0x80000000
-- 
1.8.0

